wbb/modules/admin.py

```diff
@@ -217,3 +225,175 @@ async def banme(_, message: Message):
         await message.reply_text("Banned!, Joke's on you, I'm into that shit!")
     else:
         await message.reply_text("It doesn't works that way mate.")
+
+# Delete messages
+
+
+@app.on_message(cust_filter.command(commands=("del")))
+async def delete(_, message: Message):
+    admins = await list_admins(message.chat.id)
+    chat_id = message.chat.id
+    from_user_id = message.from_user.id
+
+    if message.from_user.id in admins \
+            or message.from_user.id in SUDO:
+        if (await app.get_chat_member(chat_id,
+                                      from_user_id)).can_delete_messages \
+            or (await app.get_chat_member(chat_id, from_user_id)).status \
+            == 'creator' \
+                or message.from_user.id in SUDO:
+            await message.reply_to_message.delete()
+            await message.delete()
+    else:
+        await message.reply_text("You Don't Have Enough Permissions,"
+                                 + " Consider Deleting Yourself!")
+
```

Comment:
why the function `list_admins` does not include creator? do you need to distinguish them in `admin.py`? If not then you may append creator to the admin list, or even append members of SUDO to it.

Reply:
`.can_delete_messages` is only available to admin/creator (for groups), then you can avoid if statement inside if statement by doing something like this.

not sure if I missed anything, is it really necessary to check if the user in SUDO twice?
 
```
@app.on_message(cust_filter.command(commands=("del")))
async def delete(_, message: Message):
    admins = await list_admins(message.chat.id)
    chat_id = message.chat.id
    from_user_id = message.from_user.id
    msg_text_no_permission = ("You don't have enough permissions, "
                              "consider deleting yourself")
    if not (from_user_id in admins or from_user_id in SUDO):
        await message.reply_text(msg_text_no_permission)
        return
    if (await app.get_chat_member(chat_id, from_user_id)).can_delete_messages:
        await message.reply_to_message.delete()
        await message.delete()
```

Reply:
.can_delete_messages is only available for admins
for creator, there is no permissions listed anywhere, he can do anything that's why i have to filter creator too
if i don't use sudo for the second time. i will append sudo users in list_admin

Reply:
ok, maybe just leave it